File: LeetCode/7/7. Reverse Integer.cpp
#define INT_MAX 2147483647

class Solution {
public:
    int reverse(int x) {
        bool sign;
        int output;

        if (x <= -INT_MAX)
            return 0;

        sign = (x < 0) ? true : false;
        x = (x < 0) ? -x : x;
        output = 0;
        while (x > 0) {
            if (output > INT_MAX / 10)
                return 0;

            output = output * 10 + (x % 10);
            x /= 10;
        }
        if (sign)
            return -output;
        else
            return output;
    }
};

File: LeetCode/7/7. Reverse Integer.java
import java.lang.*;

class Solution {
    public int reverse(int x) {
        if (x < -Integer.MAX_VALUE || x > Integer.MAX_VALUE)
        {
            return 0;
        }

        int sign = 1;
        if (x < 0)
        {
            x = -x;
            sign = -1;
        }

        int i;
        String strx = Integer.toString(x);
        String rev_x = "";
        for (i = strx.length() - 1; i >= 0; i--)
        {
            rev_x += strx.charAt(i);
        }
        try
        {
            return sign * Integer.parseInt(rev_x);
        }
        catch (Exception e)
        {
            return 0;
        }

    }
}